R/map_estimate.R

```diff
@@ -46,62 +39,67 @@ map_estimate <- function(x, ...) {
 }
 
 
+
+
+#' @export
+print.MAP <- function(x, ...) {
+  cat(sprintf("MAP = %.2f", x))
+}
+
+
+
+
+
 #' @rdname map_estimate
 #' @export
-map_estimate.numeric <- function(x, precision = 2^10, density = FALSE, ...) {
+map_estimate.numeric <- function(x, precision = 2^10, ...) {
   d <- stats::density(x, n = precision)
 
   hdp_x <- d$x[which.max(d$y)]
   hdp_y <- max(d$y)
 
-  out <- data.frame(
-    "MAP" = hdp_x,
-    "MAP_density" = hdp_y
-  )
+  out <- hdp_x
+  attr(out, "MAP_density") <- hdp_y
```

Comment:
here

Reply:
ok, I see.
I fixed the test now. For vectors, or in general objects with attributes, simply wrap the function to test in `dput()` and the result goes into the test-file.

``` r
library(bayestestR)
dput(map_estimate(rnorm_perfect(1000)))
#> structure(-0.00368006135440124, MAP_density = 0.389791625903652, class = c("numeric", 
#> "MAP"))
```

Reply:
I hope this attribute won't create problems for other use cases apart from testing 🤞

Reply:
No, except you compare objects with `all.equal()` somewhere ;-) Else, attributes are pretty harmless. Even such harmless, that they're sometimes simply forgotten and dropped (e.g. after using `base::subset()`).

Reply:
Every class-_attribute_ is an _attribute_ as well. :-D